app/src/main/java/com/bignerdranch/android/criminalintent/CrimeAdapter.kt

```diff
@@ -0,0 +1,45 @@
+package com.bignerdranch.android.criminalintent
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import android.widget.Toast
+import androidx.recyclerview.widget.RecyclerView
+import com.bignerdranch.android.criminalintent.models.Crime
+
+class CrimeAdapter(var crimes: List<Crime>) : RecyclerView.Adapter<CrimeAdapter.CrimeHolder>() {
+
+  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CrimeHolder {
+    val view = LayoutInflater.from(parent.context).inflate(R.layout.list_item_crime, parent, false)
+    return CrimeHolder(view)
+  }
+
+  override fun onBindViewHolder(holder: CrimeHolder, position: Int) {
+    val crime = crimes[position]
+    holder.bind(crime)
+  }
+
+  override fun getItemCount(): Int = crimes.size
+
+  inner class CrimeHolder(view: View) : RecyclerView.ViewHolder(view), View.OnClickListener {
+    private lateinit var crime: Crime
```

Comment:
I think that's not good because it could be null. Furthermore, you don't need it